Nexogen.Libraries.Metrics.Prometheus.AspCore/PrometheusExtensions.cs

```diff
@@ -26,14 +31,34 @@ public static IServiceCollection AddPrometheus(this IServiceCollection services)
         /// Add Prometheus Metrics support to the application.
         /// </summary>
         /// <param name="builder"></param>
+        /// <param name="getHttpPath">Function to retrieve metric path from http context</param>
         /// <returns></returns>
         public static IApplicationBuilder UsePrometheus(this IApplicationBuilder builder)
         {
-            builder.UseMiddleware<CollectMetricsMiddleware>();
-            return builder.Map("/metrics", cfg =>
+            return builder.UseMiddleware<CollectMetricsMiddleware>()
+                .Map("/metrics", cfg =>
+                {
+                    cfg.UseMiddleware<ServeMetricsMiddleware>();
+                });
+        }
+
+        /// <summary>
+        /// Gets the last <see cref="RouteBase"/> that
+        /// matched the request. We use the route base because the <see cref="IRouter"/>
+        /// doesn't expose the template text.        
+        /// </summary>
+        /// <param name="routers">List of routers that matched the request</param>
+        /// <returns></returns>
+        internal static RouteBase FindLast(this IList<IRouter> routers)
```

Comment:
One more last thing. Could you please also move this method to `CollectMetricsMiddleware`? I feel it rather belongs there as a private method.